src/ir_passes/bundle_elim.rs

```diff
@@ -217,20 +217,52 @@ impl BundleElim {
             self.context.get_mut(cidx).extend(pl);
         }
 
-        // compile local ports and adds them to the context
-        for idx in comp
-            .ports()
+        // generates the local map for this component
+        self.local_map = comp
+            .cmds
+            .iter()
+            .filter_map(|cmd| {
+                let Command::Connect(con) = cmd else { return None };
+                let Connect { src, dst, .. } = con;
+
+                // need to check validity here because already deleted ports are still in the connect commands
+                if comp.ports().is_valid(dst.port)
+                    && comp.get(dst.port).is_local()
+                {
+                    let dst_start =
+                        dst.start.as_concrete(comp).unwrap() as usize;
```

Comment:
Can we expose a method on `ExprIdx` to do this in one go? Maybe a `Into` implementation so we could write:
```
let dst_start: usize = dst.start.try_into()
```
One problem is that the `Into` implementation could panic but that can already happen here?

Reply:
Can't write an into because it requires a context to resolve, the only thing that would be different from `as_concrete` would be the cast to `usize` which I'm not sure is that useful.

Reply:
Gotcha!